rspace/src/main/scala/coop/rchain/rspace/ReportingRspace.scala

```diff
@@ -105,31 +125,23 @@ class ReportingRspace[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C, P,
   implicit protected[this] override lazy val MetricsSource: Metrics.Source =
     Metrics.Source(RSpaceMetricsSource, "reporting")
 
-  /**
-    * in order to distinguish the system deploy(precharge and refund) in the a normal user deploy
-    * It might be more easily to analyse the report with data structure
-    * Seq(Seq[ReportingEvent](Precharge), Seq[ReportingEvent](userDeploy), Seq[ReportingEvent](Refund))
-    * It would be seperated by the softcheckpoint creation.
-    */
-  val report: SyncVar[Seq[Seq[ReportingEvent]]] = create[Seq[Seq[ReportingEvent]]](Seq.empty)
-  val softReport: SyncVar[Seq[ReportingEvent]]  = create[Seq[ReportingEvent]](Seq.empty)
-
   private def collectReport =
     for {
       sReport <- getSoftReport
-      _ = if (sReport.nonEmpty) {
-        report.update(s => s :+ sReport)
-        softReport.update(_ => Seq.empty[ReportingEvent])
-      }
+      _ <- if (sReport.nonEmpty) for {
+            _ <- report.update(s => s :+ sReport)
+            _ <- softReport.update(_ => Seq.empty[ReportingEvent])
+          } yield ()
+          else Unit.pure[F]
```

Comment:
This can be written as 
```scala
_ <- (report.update(s => s :+ sReport) >> softReport.update(_ => Seq.empty[ReportingEvent])).whenA(sReport.nonEmpty)
```
the whole collectReport can be 
```scala
softReport.get.flatMap{ sReport => 
  (report.update(s => s :+ sReport) >> softReport.update(_ => Seq())).whenA(sReport.nonEmpty)
}
```

Reply:
">>" - this syntax is unfamiliar to me. Thanks! I fixed this.